mantidproject/mantid
# Replace Instrument 1.0 access with ComponentInfo/DetectorInfo

This PR involves the following work items **NB: A lot of the changes are files which have been removed.**:
* remove gui actors and replace with `ComponentInfo`
* remove references to `Instrument`
* remove references to `IObjComponent` `IComponentAssembly` `ObjComponentAssembly` `ComponentAssembly`

as far as possible use `ComponentInfo` and/or `DetectorInfo` for instrument access.

**To test:**

* Thorough code review
* Load **Instrument View** and ensure a wide selection of instruments are displayed correctly:
  * ensure all detectors are displayed correctly in each view (unwrapped views and FULL3D)
  * ensure picking, masking and selection all work correctly
  * ensure the instrument tree displays the correct detectors (can be verified using pick tab).

Fixes #21339. 

**Release Notes** 
*Does not need to be in the release notes. The performance improvements will be included in the release notes as part of the main feature branch.*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
